### PR TITLE
ue5.1

### PR DESCRIPTION
**Pull request by TheEmidee**

### CODE REVIEW

File: Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetDataUsingActor.cpp [hunk 1]
```diff
@@ -35,20 +35,20 @@ void UGASExtAT_WaitTargetDataUsingActor::Activate()
 
 void UGASExtAT_WaitTargetDataUsingActor::OnTargetDataReplicatedCallback( const FGameplayAbilityTargetDataHandle & data, FGameplayTag activation_tag )
 {
-    check( IsValid( AbilitySystemComponent ) );
+    check( AbilitySystemComponent != nullptr );
```
Comment (VectorGamez) [inline Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetDataUsingActor.cpp:38]: why not use AbilitySystemComponent.IsValid()?

File: Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetDataUsingActor.cpp [hunk 2]
```diff
@@ -82,14 +82,14 @@ void UGASExtAT_WaitTargetDataUsingActor::OnTargetDataReplicatedCancelledCallback
 
 void UGASExtAT_WaitTargetDataUsingActor::OnTargetDataReadyCallback( const FGameplayAbilityTargetDataHandle & data )
 {
-    check( IsValid( AbilitySystemComponent ) );
+    check( AbilitySystemComponent != nullptr );
```
Comment (VectorGamez) [inline Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetDataUsingActor.cpp:85]: same

File: Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetDataUsingActor.cpp [hunk 3]
```diff
@@ -119,9 +119,9 @@ void UGASExtAT_WaitTargetDataUsingActor::OnTargetDataReadyCallback( const FGamep
 
 void UGASExtAT_WaitTargetDataUsingActor::OnTargetDataCancelledCallback( const FGameplayAbilityTargetDataHandle & Data )
 {
-    check( IsValid( AbilitySystemComponent ) );
+    check( AbilitySystemComponent != nullptr );
```
Comment (VectorGamez) [inline Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetDataUsingActor.cpp:122]: same

File: Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetDataUsingActor.cpp [hunk 4]
```diff
@@ -141,7 +141,7 @@ void UGASExtAT_WaitTargetDataUsingActor::OnTargetDataCancelledCallback( const FG
 
 void UGASExtAT_WaitTargetDataUsingActor::ExternalConfirm( const bool end_task )
 {
-    check( IsValid( AbilitySystemComponent ) );
+    check( AbilitySystemComponent != nullptr );
```
Comment (VectorGamez) [inline Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetDataUsingActor.cpp:144]: same

File: Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetDataUsingActor.cpp (hunk 1 above)
Comment (Frickerson) [inline Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetDataUsingActor.cpp:38]: What is the reason you changed from the isvalid check to checking nullptr?

**Review by Frickerson — COMMENTED** [review]
Seems like it does not compile. Are some changes missing, or is it a UE5.1 thing?

File: Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetData.cpp [hunk 5]
```diff
@@ -82,14 +82,14 @@ void UGASExtAT_WaitTargetData::OnTargetDataReplicatedCallback( const FGameplayAb
...
-    check( IsValid( AbilitySystemComponent ) );
+    check( AbilitySystemComponent != nullptr );
     if ( Ability == nullptr )
     {
         return;
     }
 
     FScopedPredictionWindow scoped_prediction(
-        AbilitySystemComponent,
+        AbilitySystemComponent.Get(),
```
Comment (Frickerson) [inline Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetData.cpp:92]: Does not compile, header change missing?

File: Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetDataUsingActor.cpp [hunk 6]
```diff
@@ -82,14 +82,14 @@ void UGASExtAT_WaitTargetDataUsingActor::OnTargetDataReplicatedCancelledCallback
...
-    check( IsValid( AbilitySystemComponent ) );
+    check( AbilitySystemComponent != nullptr );
     if ( Ability == nullptr )
     {
         return;
     }
 
     FScopedPredictionWindow scoped_prediction(
-        AbilitySystemComponent,
+        AbilitySystemComponent.Get(),
```
Comment (Frickerson) [inline Source/GASExtensions/Private/Tasks/GASExtAT_WaitTargetDataUsingActor.cpp:92]: Does not compile, header change missing?

File: Source/GASExtensions/Private/Tasks/GASExtAT_WaitAttributeChangeWithValues.cpp [hunk 7]
```diff
@@ -114,7 +115,7 @@ void UGASExtAT_WaitAttributeChangeWithValues::OnAttributeChange( const FOnAttrib
 
 UAbilitySystemComponent * UGASExtAT_WaitAttributeChangeWithValues::GetFocusedASC()
 {
-    return ExternalOwner ? ExternalOwner : AbilitySystemComponent;
+    return ExternalOwner != nullptr ? ExternalOwner.Get() : AbilitySystemComponent.Get();
```
Comment (Frickerson) [inline Source/GASExtensions/Private/Tasks/GASExtAT_WaitAttributeChangeWithValues.cpp:118]: does not compile, header change missing?

File: Source/GASExtensions/Private/Tasks/GASExtAT_WaitAbilityEnd.cpp [hunk 8]
```diff
@@ -20,7 +20,7 @@ void UGASExtAT_WaitAbilityEnd::Activate()
 
     SetWaitingOnAvatar();
 
-    if ( !IsValid( OptionalASC ) )
+    if ( OptionalASC != nullptr )
```
Comment (Frickerson) [inline Source/GASExtensions/Private/Tasks/GASExtAT_WaitAbilityEnd.cpp]: why not use .IsValid()?

File: Source/GASExtensions/Private/Tasks/GASExtAT_ServerWaitClientTargetData.cpp [hunk 9]
```diff
@@ -39,7 +39,7 @@ void UGASExtAT_ServerWaitClientTargetData::OnTargetDataReplicatedCallback( const
 
 void UGASExtAT_ServerWaitClientTargetData::OnDestroy( const bool ability_ended )
 {
-    if ( IsValid( AbilitySystemComponent ) )
+    if ( AbilitySystemComponent.IsValid() )
```
Comment (Frickerson) [inline Source/GASExtensions/Private/Tasks/GASExtAT_ServerWaitClientTargetData.cpp:42]: Does not compile, header change missing?

File: Source/GASExtensions/Private/GameplayCues/GASExtGameplayCueManager.cpp [hunk 10]
```diff
@@ -137,14 +137,13 @@ void UGASExtGameplayCueManager::LoadAlwaysLoadedCues()
...
-    auto * runtime_gameplay_cue_set = GetRuntimeCueSet();
-    if ( runtime_gameplay_cue_set )
+    if ( const auto * runtime_gameplay_cue_set = GetRuntimeCueSet() )
     {
         runtime_gameplay_cue_set->GetSoftObjectPaths( cue_paths );
     }
 
     FAssetBundleData bundle_data;
-    bundle_data.AddBundleAssets( GASExtGameplayCueManagerCvars::UGASExtAssetManager_LoadStateClient, cue_paths );
+    bundle_data.AddBundleAssetsTruncated( GASExtGameplayCueManagerCvars::UGASExtAssetManager_LoadStateClient, cue_paths );
```
Comment (Frickerson) [inline Source/GASExtensions/Private/GameplayCues/GASExtGameplayCueManager.cpp:146]: Does not compile, header changes missing?

File: Source/GASExtensions/Private/Components/GASExtAbilitySystemComponent.cpp [hunk 11]
```diff
@@ -1133,11 +1132,11 @@ bool UGASExtAbilitySystemComponent::ServerCurrentMontageSetNextSectionNameForMes
 void UGASExtAbilitySystemComponent::ServerCurrentMontageJumpToSectionNameForMesh_Implementation( USkeletalMeshComponent * mesh, UAnimMontage * client_anim_montage, const FName section_name )
 {
     auto * anim_instance = GetMeshAnimInstance( mesh );
-    auto & anim_montage_info = GetLocalAnimMontageInfoForMesh( mesh );
+    const auto & anim_montage_info = GetLocalAnimMontageInfoForMesh( mesh );
 
     if ( anim_instance )
     {
-        auto * current_anim_montage = anim_montage_info.LocalMontageInfo.AnimMontage;
+        const auto current_anim_montage = anim_montage_info.LocalMontageInfo.AnimMontage;
```
Comment (Frickerson) [inline Source/GASExtensions/Private/Components/GASExtAbilitySystemComponent.cpp:1140]: const auto *

File: Source/GASExtensions/Private/Components/GASExtAbilitySystemComponent.cpp [hunk 12]
```diff
@@ -1160,11 +1159,11 @@ bool UGASExtAbilitySystemComponent::ServerCurrentMontageJumpToSectionNameForMesh
 void UGASExtAbilitySystemComponent::ServerCurrentMontageSetPlayRateForMesh_Implementation( USkeletalMeshComponent * mesh, UAnimMontage * client_anim_montage, const float play_rate )
 {
     auto * anim_instance = GetMeshAnimInstance( mesh );
-    auto & anim_montage_info = GetLocalAnimMontageInfoForMesh( mesh );
+    const auto & anim_montage_info = GetLocalAnimMontageInfoForMesh( mesh );
 
     if ( anim_instance )
     {
-        auto * current_anim_montage = anim_montage_info.LocalMontageInfo.AnimMontage;
+        const auto current_anim_montage = anim_montage_info.LocalMontageInfo.AnimMontage;
```
Comment (Frickerson) [inline Source/GASExtensions/Private/Components/GASExtAbilitySystemComponent.cpp:1167]: const auto *

**Review by Frickerson — DISMISSED** [review]

**Review by Frickerson — DISMISSED** [review]
:LGTM:

**Review by Frickerson — APPROVED** [review]

**Pull request merged**